### DISCUSSION TITLE
De-anonymization based on a unique port number

### DISCUSSION THREAD

**Discussion by tengkuizdihar** [PurpleI2P/i2pd#2000] [Q&A] [answered]
This was a question that popped out in the unofficial i2p matrix. The question as follows.

```
What information will the attackers get if you are the only person in the whole i2p network that uses port 1234? Will you be able to link 1234 with the packets I'm sending? The content and or the destination?
```

I'm on the understanding that having a unique port means nothing to the factor in which you're anonymized. Because, ports and IP address (192.1.1.1:1234) come in together and should've been anonymized by the i2p protocol. What do you think, Am I correct?

**Comment by Vort** [comment]
If you say someone that you use specific unique port, then it is possible to obtain your IP address.
Nothing more than that if I understand correctly.
I2P addresses (destinations) and IP addresses are handled by different protocols, so I think they are isolated enough.

**Reply by tengkuizdihar** [reply]
will the attacker be able to correlate that with who I'm talking to and the content of the packet?

**Reply by Vort** [reply]
I can't understand your questions, sorry.
Maybe someone else will clarify situation more.

**Reply by tengkuizdihar** [reply]
let me clarify, what I meant was,
if someone know my unique port, therefore also my IP address, is it possible for them to also know:
* what eepsite I'm accessing
* the information that I'm giving to that eepsite, and vice versa

**Reply by Vort** [reply, marked as answer]
As far as I know, anonymization is based on amounts of hops.
0 or 1 hops gives low anonymity, 3 hops gives high anonymity.
Only hop which you are directly connected to, knows your IP address.
Other hops know nothing about it.

Conversation with eepsite should be encrypted anyway.
